tests/ttnn/unit_tests/operations/test_pad.py

```diff
@@ -82,17 +82,18 @@ def run_pad_rm_sharded(device, n, c, h, w, padding, torch_padding, value):
         memory_config=ttnn.L1_MEMORY_CONFIG,
     )
 
-    H_pad = padding[1][1]
-    W_pad = padding[2][1]
+    n_unpadded = n
+    c_unpadded = c + padding[0][1]
+    h_unpadded = h + padding[1][1]
 
     # shard config
-    num_cores_x = 6
+    num_cores_x = 8
     num_cores_y = 8
+    shard_h = (n * c * h + (num_cores_x * num_cores_y) - 1) // (num_cores_x * num_cores_y)
     grid_size = ttnn.CoreGrid(y=num_cores_y, x=num_cores_x)
-
     grid_coord = ttnn.CoreCoord(grid_size.x - 1, grid_size.y - 1)
     shard_grid = ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), grid_coord)})
-    shard_spec = ttnn.ShardSpec(shard_grid, (h, w), ttnn.ShardOrientation.ROW_MAJOR, False)
+    shard_spec = ttnn.ShardSpec(shard_grid, (shard_h, w), ttnn.ShardOrientation.COL_MAJOR, False)
     sharded_mem_config = ttnn.MemoryConfig(
```

Comment:
Rather than replace this test, should we instead be adding a new one?

Reply:
yeah, I can add more args to this test to cover the original